src/gwt/src/org/rstudio/studio/client/panmirror/theme/PanmirrorThemeCreator.java

```diff
@@ -37,7 +38,12 @@ public static PanmirrorTheme themeFromEditorTheme(AceTheme aceTheme, UserPrefs p
       theme.darkMode = aceTheme.isDark();
       theme.solarizedMode = aceTheme.isSolarizedLight();
       
-      theme.cursorColor = DomUtils.extractCssValue("ace_cursor", "color");
+      // get cursor color (guard against themes that don't set ace_cursor color)
+      String defaultCursorColor = aceTheme.isDark() ? "white" : "black";
+      String cursorColor = DomUtils.extractCssValue("ace_cursor", "color");
+      theme.cursorColor = StringUtil.isNullOrEmpty(cursorColor)
+            ? cursorColor
```

Comment:
Is this right? It looks like it will use the `cursorColor` only when it's empty.

Reply:
Oops; you are correct. Fixed up with a slightly different approach now.